### ISSUE TITLE
Highlight locations on overmap that were revealed from reading a map by using the map again

### ISSUE THREAD

**Issue by Chadtholomew** [CleverRaven/Cataclysm-DDA#63380]
### Is your feature request related to a problem? Please describe.

When a lot of the overmap is revealed from various different sources it makes it difficult to quickly identify interesting POIs unless they are in the zones that are still hidden or you know exactly what you are looking for.

### Solution you would like.

I would like to use a map to highlight only the locations that the map provided and be able to quickly hop between those areas. Having this option allows me to quickly see what was on that map and can help plan my routes. From a role-playing standpoint It would also give a reason to hold onto precious information like that and give more use for items like journals which can hold them. An option to scan them into your tablet PC could work and be immersive too.

**Comment by ZhilkinSerg** [comment]
Did you know you can leave notes, mark explored tiles and use filter to quickly search for specific terrains on an overmap?

**Comment by Chadtholomew** [comment]
> Did you know you can leave notes, mark explored tiles and use filter to quickly search for specific terrains on an overmap?

Yes, but that doesn't help me to identify what POIs have been added from map use. I can search for "gun store", because that's a POI that I know exists, but I can't search for everything that a survivor's map would have added because I don't know all the things that it can add, and I can't account for things that are added into the game in the future either. When a map drops and things are added to your map, that's kind of an indicator that you should go visit that area, however if I've climbed radio towers and hacked lab computers or just travelled a lot, much of the map is revealed and those maps become less useful to quickly plan a looting run because much of the map fog is gone, making it more difficult to quickly identify POIs.

**Comment by ZhilkinSerg** [comment]
So, your character have learned there are several gun stores on current overmap and their locations are now marked on the overmap. I don't see how it would help to know what was the source of this knowledge (i.e. was it specific map revealing item or you stumbled upon some location by yourself)?

Say, you store all the map revealing items in your inventory and when activating one of the maps you see something like this:

![image](https://user-images.githubusercontent.com/16213433/216149248-443d8092-0016-46c3-a5a6-35955923074f.png)

How is it helpful? You can simply search for gun stores on overmap menu and jump through them adding notes when necessary. W/o changing the overmap UI it would be easy to create auto notes to all overmap cells revealed by map revealing items, but I lack to see which information in autonote could be useful. Also we could do so map revealing items could create dynamic quests to visit revealed POI (so we had target arrows to know where to move), but most of these locations aren't really unique and it does not really matter if you visit some generic gun store or skip it.

**Comment by Chadtholomew** [comment]
> How is it helpful? You can simply search for gun stores on overmap menu and jump through them adding notes when necessary.

Yes, you can search for gun stores on the map. In my previous comment I explained that:
> I can search for "gun store", because that's a POI that I know exists, but I can't search for everything that a survivor's map would have added because I don't know all the things that it can add, and I can't account for things that are added into the game in the future either.


> Also we could do so map revealing items could create dynamic quests to visit revealed POI (so we had target arrows to know where to move), but most of these locations aren't really unique and it does not really matter if you visit some generic gun store or skip it.

I had a similar idea that would be a byproduct of this proposed new system. Similar to how certain random overmap locations are only added to your map notes by being in close proximity to them (casings, sand patch, burned area, college students, portal, etc), some hidden (or secret) locations could be revealed by using some of those maps. Downloading map data or looking from a radio tower shouldn't show you a hidden location from such a distance. One could find a survivor's note that gives you the location of a house with a basement containing tailoring supplies, or as a quest reward an NPC can tell you of a secret trapdoor in a derelict property, or you hear coordinates over the radio for an overgrown LMOE shelter that would be difficult to spot unless you were physically on top of it. 

But, that's a different feature request for a different time, for now it would be helpful just to see what locations a map contains.

**Comment by Vaskritaya** [comment]
> W/o changing the overmap UI it would be easy to create auto notes to all overmap cells revealed by map revealing items, but I lack to see which information in autonote could be useful. 

There are some maps like bookmark_note_catalytic which could heavily benefit from any sort of further guidance. This map in particular reveals a gray tile somewhere in the sea of undiscovered gray tiles, and it's quite hard to search for.

**Comment by MelacholicMimic** [comment]
The dragon lair map from magiclysm would also benefit since it just reveals 2x2 swamp tiles, making it impossible to find unless it's in an unexplored location.
